server/src/main/java/io/deephaven/server/table/ops/AggSpecAdapter.java

```diff
@@ -95,27 +133,23 @@ private static io.deephaven.api.agg.spec.AggSpecApproximatePercentile adapt(
     }
 
     private static io.deephaven.api.agg.spec.AggSpecCountDistinct adapt(AggSpecCountDistinct countDistinct) {
-        return countDistinct.hasCountNulls() ? AggSpec.countDistinct(countDistinct.getCountNulls())
-                : AggSpec.countDistinct();
+        return AggSpec.countDistinct(countDistinct.getCountNulls());
     }
 
     private static io.deephaven.api.agg.spec.AggSpecDistinct adapt(AggSpecDistinct distinct) {
-        return distinct.hasIncludeNulls() ? AggSpec.distinct(distinct.getIncludeNulls()) : AggSpec.distinct();
+        return AggSpec.distinct(distinct.getIncludeNulls());
     }
 
     private static io.deephaven.api.agg.spec.AggSpecFormula adapt(AggSpecFormula formula) {
-        return formula.hasParamToken() ? AggSpec.formula(formula.getFormula(), formula.getParamToken())
-                : AggSpec.formula(formula.getFormula());
+        return AggSpec.formula(formula.getFormula(), formula.getParamToken());
     }
 
     private static io.deephaven.api.agg.spec.AggSpecMedian adapt(AggSpecMedian median) {
-        return median.hasAverageEvenlyDivided() ? AggSpec.median(median.getAverageEvenlyDivided()) : AggSpec.median();
+        return AggSpec.median(median.getAverageEvenlyDivided());
     }
 
     private static io.deephaven.api.agg.spec.AggSpecPercentile adapt(AggSpecPercentile percentile) {
-        return percentile.hasAverageEvenlyDivided()
-                ? AggSpec.percentile(percentile.getPercentile(), percentile.getAverageEvenlyDivided())
-                : AggSpec.percentile(percentile.getPercentile());
+        return AggSpec.percentile(percentile.getPercentile(), percentile.getAverageEvenlyDivided());
     }
 
     private static SortColumn adapt(AggSpecSortedColumn sortedColumn) {
```

Comment:
Still some shared messages. Should we change that?

Reply:
I vote no.